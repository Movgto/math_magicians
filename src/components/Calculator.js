import '../stylesheets/calculator.scss';
import { useState } from 'react';
import calculate from '../logic/calculate';
import Screen from './Screen';

function Calculator() {
  const [state, setState] = useState({
    total: null,
    next: null,
    operation: null,
  });
  return (
    <div id="calculator" data-testid="calculator">
      <Screen total={state.total} next={state.next} />
      <div className="row">
        <button className="button" type="button" onClick={() => setState(calculate(state, 'AC'))}>AC</button>
        <button className="button" type="button" onClick={() => setState(calculate(state, '+/-'))}>+/-</button>
        <button className="button" type="button" onClick={() => setState(calculate(state, '%'))}>%</button>
        <button className="button operator" type="button" onClick={() => setState(calculate(state, 'รท'))}>รท</button>
      </div>
      <div className="row">
        <button className="button" type="button" onClick={() => setState(calculate(state, '7'))}>7</button>
        <button className="button" type="button" onClick={() => setState(calculate(state, '8'))}>8</button>
        <button className="button" type="button" onClick={() => setState(calculate(state, '9'))}>9</button>
        <button className="button operator" type="button" onClick={() => setState(calculate(state, 'x'))}>x</button>
      </div>
      <div className="row">
        <button className="button" type="button" onClick={() => setState(calculate(state, '4'))}>4</button>
        <button className="button" type="button" onClick={() => setState(calculate(state, '5'))}>5</button>
        <button className="button" type="button" onClick={() => setState(calculate(state, '6'))}>6</button>
        <button className="button operator" type="button" onClick={() => setState(calculate(state, '+'))}>+</button>
      </div>
      <div className="row">
        <button className="button" type="button" onClick={() => setState(calculate(state, '1'))}>1</button>
        <button className="button" type="button" onClick={() => setState(calculate(state, '2'))}>2</button>
        <button className="button" type="button" onClick={() => setState(calculate(state, '3'))}>3</button>
        <button className="button operator" type="button" onClick={() => setState(calculate(state, '-'))}>-</button>
      </div>
      <div className="row">
        <button className="button double-width" type="button" onClick={() => setState(calculate(state, '0'))}>0</button>
        <button className="button" type="button" onClick={() => setState(calculate(state, '.'))}>.</button>
        <button className="button" type="button" onClick={() => setState(calculate(state, '='))}>=</button>
      </div>
    </div>
  );
}

export default Calculator;
