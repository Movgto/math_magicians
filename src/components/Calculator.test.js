import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Calculator from './Calculator';

describe('Snapshot', () => {
  it('renders properly', () => {
    const tree = renderer
      .create(<Calculator />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Calculator Component', () => {
  it('renders without errors', () => {
    render(<Calculator />);
  });

  // Test button clicks and interactions
  it('updates state on button click', () => {
    const { getByText } = render(<Calculator />);
    const seven = getByText('7'); // Change this to the label of the button you want to test
    const plus = getByText('+');
    const equals = getByText('=');
    fireEvent.click(seven);
    fireEvent.click(plus);
    fireEvent.click(seven);
    fireEvent.click(equals);
    getByText('14');
    // Add assertions to check if the state has been updated as expected
  });

  // Add more test cases for other interactions
});
