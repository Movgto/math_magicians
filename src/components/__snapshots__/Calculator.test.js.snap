// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot renders properly 1`] = `
<div
  data-testid="calculator"
  id="calculator"
>
  <div
    id="screen"
  >
    0
  </div>
  <div
    className="row"
  >
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="button operator"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
  </div>
  <div
    className="row"
  >
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="button operator"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
  </div>
  <div
    className="row"
  >
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="button operator"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>
  <div
    className="row"
  >
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="button operator"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </div>
  <div
    className="row"
  >
    <button
      className="button double-width"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="button"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
